File: Code/Java-code/Part.java
import java.text.DecimalFormat;
public class Part {
    DecimalFormat df = new DecimalFormat("###,###.##");

    private String name;
    private double price;
    private int id;
    
    public static double totalPartCost = 0;

    private ServiceAppointment serviceAppointment;

    public Part(String part, double price, ServiceAppointment serviceAppointment)
    {
        this.name = part;
        this.price = price;
        this.serviceAppointment = serviceAppointment;

        totalPartCost += price;
    }

    public String getName() {
        return this.name;
    }

    public int getId() {
        return this.id;
    }

    public double getPrice() {
        return this.price;
    }

    public void setId(int id) {
        this.id = id;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setPrice(double price) {
        this.price = price;
    }

    public static String getTotalPartCost() {
        return "Total part cost is = "+ totalPartCost;
    }

    public static void ResetTotalPartCost()
    {
        totalPartCost = 0;
    }

    @Override
    public String toString() {
        return this.name + " = " + df.format(this.price) + " for " + serviceAppointment.getCar().getRegistration();
    }
}

File: Code/PHP-Code/src/Part.php
<?php
class Part{

    private String $part;
    private float $price;
    private int $id;

    public static float $totalPartCost = 0;

    public function __construct(String $part, float $price )
    {
        $this->part = $part;
        $this->price = $price;

        self::$totalPartCost += $price;

    }

    public function setId(int $id): void
    {
        $this->id = $id;
    }


    public function setPart(string $part): void
    {
        $this->part = $part;
    }

    public function setPrice(float $price): void
    {
        $this->price = $price;
    }

    public function getId(): int
    {
        return $this->id;
    }


    public function getPart(): string
    {
        return $this->part;
    }


    public function getPrice(): float
    {
        return $this->price;
    }

    public static function getTotalPartCost(): float|String
    {
        return "\nTotal part cost is = ". self::$totalPartCost;
    }

    public static function setTotalPartCost(): void
    {
        self::$totalPartCost = 0;
    }

    public function __toString(): string
    {
        return "\n".$this->part . " = " .$this->price;
    }


}